addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -31,11 +31,13 @@ function factory(dependencies) {
                 limit,
             }));
             this.update({ isLoadingMore: false });
-            if (fetchedMessages.length < limit) {
-                this.update({ isAllHistoryLoaded: true });
-            }
-            for (const threadView of this.threadViews) {
-                threadView.addComponentHint('more-messages-loaded', { fetchedMessages });
+            if (fetchedMessages) {
+                if (fetchedMessages.length < limit) {
+                    this.update({ isAllHistoryLoaded: true });
+                }
+                for (const threadView of this.threadViews) {
+                    threadView.addComponentHint('more-messages-loaded', { fetchedMessages });
+                }
```

Comment:
Code duplication, all `loadXMessages()` method use `_loadMessages()`, so this private method should add these component hints.

Reply:
It's not the same hint, `messages-loaded', `new-messages-loaded` & `more-messages-loaded`

Reply:
Seems to work, though

Reply:
Ok, didn't see it was a different hint, just ignore this comment.